tests/test_utils_strict_dataclass.py

```diff
@@ -29,6 +29,24 @@ def strictly_positive(value: int):
         raise ValueError(f"Value must be strictly positive, got {value}")
 
 
+def dtype_validation(value):
+    "Torch isn't installed in runners so we will check only for `string`"
+    if not isinstance(value, str):
+        raise ValueError(f"Value must be string ot `torch.dtype` but got {value}")
+
+    if isinstance(value, str) and value not in ["float32", "bfloat16", "float16"]:
+        raise ValueError(f"Value must be one of `[float32, bfloat16, float16] but got {value}")
+
+
+@strict
+@dataclass
+class ConfigForwardRef:
+    model_type: str
```

Comment:
```suggestion
```

no need to test `str` in this class. We only want to test the "forward ref"-related logic